src/Backups/BackupImpl.cpp

```diff
@@ -144,6 +144,7 @@ void BackupImpl::open(const ContextPtr & context)
         if (!uuid)
             uuid = UUIDHelpers::generateV4();
         lock_file_name = use_archive ? (archive_params.archive_name + ".lock") : ".lock";
+        lock_file_before_first_file_checked = false;
```

Comment:
Why can we do this just in constructor of BackupImpl? I afraid we can start to use  `BackupImpl::writeFile` when `open_mode != OpenMode::WRITE`

Reply:
Files can be written only in `OpenMode::WRITE`.